app/components/channel_list/categories/body/index.ts

```diff
@@ -87,7 +88,21 @@ const enhance = withObservables(['category'], ({category, locale, database}: {ca
         switchMap((c) => getSortedIds(database, c, locale)),
     );
 
+    let limit = of$('');
+    if (category.type === 'direct_messages') {
+        limit = database.get<PreferenceModel>(PREFERENCE).
+            query(
+                Q.where('category', Preferences.CATEGORY_SIDEBAR_SETTINGS),
+                Q.where('name', 'limit_visible_dms_gms'),
+            ).observe().pipe(
+                switchMap(
+                    (val) => of$(val[0].value),
```

Comment:
This should be checking if val[0] exist if not return of$('')

Also I rather do the parseInt here and have the component receive an integer.